### PR TITLE
fix: remove the constraint on branch name

### PR DESCRIPTION
**Pull request by fleboulch**
This constraint breaks the commits interaction on git branches with a `/`

## Steps to reproduce

1. create a branch with a `/` (example: `feat/my-branch`)
2. add some cool stuff and commit them
3. push and open an MR
4. in slack: `/homer review !<ID>`
5. add a new commit
6. you should not receive any message in the thread (whereas you should receive one)

### CODE REVIEW

File: src/review/commands/share/hookHandlers/pushHookHandler.ts [hunk 1]
```diff
@@ -29,10 +29,8 @@ export async function pushHookHandler(
     return;
   }
 
-  const branchName = ref.split('/').pop() as string;
```
Comment (greg0ire) [inline src/review/commands/share/hookHandlers/pushHookHandler.ts:32]: Looking at the test logs, if the branch is `foo/bar`, then `ref` is going to be `refs/branch/foo/bar`. This isn't right.

**Comment by greg0ire** [comment]
Is it possible to add a test (or change existing tests so that they fail without your change?)

**Comment by fleboulch** [comment]
I added a new test

**Review by greg0ire — APPROVED** [review]

File: __tests__/__fixtures__/hooks/pushHookFixture.ts [hunk 2]
```diff
@@ -72,3 +72,76 @@ export const pushHookFixture = {
...
+    {
+      id: 'b6568db1bc1dcd7f8b4d5a946b0b91f9dacd7327',
+      message:
+        'Update Catalan translation to e38cb41.\n\nSee https://gitlab.com/gitlab-org/gitlab for more information',
+      title: 'Update Catalan translation to e38cb41.',
+      timestamp: '2011-12-12T14:27:31+02:00',
+      url: 'http://example.com/mike/diaspora/commit/b6568db1bc1dcd7f8b4d5a946b0b91f9dacd7327',
+      author: {
+        name: 'Jordi Mallach',
+        email: 'jordi@softcatala.org',
```
Comment (pfongkye) [inline __tests__/__fixtures__/hooks/pushHookFixture.ts:125]: Are these fictitious data?

File: src/review/commands/share/hookHandlers/pushHookHandler.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export async function pushHookHandler(
...
+  const branchName = ref.split('/').slice(2).join('/') as string;
```
Comment (pfongkye) [inline src/review/commands/share/hookHandlers/pushHookHandler.ts:32]: Is it always two words before having branch name? I suppose here we are assuming the ref will always start with something like `refs/branch`, right?

**Comment by greg0ire** [comment]
Don't mind me, just trying to trigger a pipeline with the brand new coverage action contributed in #25

**Comment by fleboulch** [comment]
@pfongkye did you have some time to check this MR?

**Comment by pfongkye** [comment]
> @pfongkye did you have some time to check this MR?

Hello,
Yes I left two comments. If you can take a look. Apart from this, LGTM.

**Comment by fleboulch** [comment]
I replied to both of them. Do you agree and resolve them or not?

**Comment by pfongkye** [comment]
> I replied to both of them. If you agree with them I can resolve them?

Sorry @fleboulch but I'm not seeing your replies for [this](https://github.com/ManoManoTech/homer/pull/15#discussion_r1634609787) and [this](https://github.com/ManoManoTech/homer/pull/15#discussion_r1637828113) :pray:

**Comment by fleboulch** [comment]
They were in pending state... :boom:

**Review by pfongkye — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @fleboulch !

**Pull request merged**